storage/directory.go

```diff
@@ -121,23 +121,80 @@ func (s *DirectoryClient) Put(ctx context.Context, input *PutDirectoryInput) err
 // DeleteDirectoryInput represents parameters to a Delete operation.
 type DeleteDirectoryInput struct {
 	DirectoryName string
+	ForceDelete   bool
 }
 
 // Delete deletes a directory on the Triton Object Storage. The directory must
 // be empty.
 func (s *DirectoryClient) Delete(ctx context.Context, input *DeleteDirectoryInput) error {
 	fullPath := path.Clean(path.Join("/", s.client.AccountName, input.DirectoryName))
 
+	if input.ForceDelete {
+		err := deleteAll(*s, ctx, input.DirectoryName)
+		if err != nil {
+			return err
+		}
+	} else {
+		err := deleteDirectory(*s, ctx, fullPath)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func deleteAll(c DirectoryClient, ctx context.Context, directoryPath string) error {
+	objs, err := c.List(ctx, &ListDirectoryInput{
+		DirectoryName: directoryPath,
+	})
+	if err != nil {
+		return err
+	}
+	for _, obj := range objs.Entries {
+		newPath := path.Clean(path.Join(directoryPath, obj.Name))
+		if obj.Type == "directory" {
```

Comment:
This isn't something specific to this PR, but I don't like the `"directory"` string here.  My gut says `obj.Type` should have its own `type` declaration and `"directory"` should be a constant someplace.  Something like:

```go
type MantaType int

const (
        UnknownType MantaType = iota
        ObjectType
        DirectoryType

        MaxTypes // Must be last
)
```

Reply:
This will be a follow up PR to change this as it's a breaking change